### ISSUE TITLE
XML-Output with exactly 3 elements per name (for phones with limitations)

### ISSUE THREAD

**Issue by shamrockDE** [Massi-X/freepbx-phonemiddleware#6]
**Is your feature request related to a problem? Please describe.**
As described in issue #3 i am using a Dect-Station m300 from Snom. The xml-phonebook for Snom needs exactly 3 elements: "Telephone", "Mobile" and "Office".

**Describe the solution you'd like**
It is very easy to implement. In the file "helpers/phoneMiddleware.php" the following code has to be added after line 255:
```
if (!$usedFields["Telephone"]) {
   $simplified_xml->writeElement('Telephone');
   $usedFields["Telephone"] = true;
}
if (!$usedFields["Mobile"]) {
   $simplified_xml->writeElement('Mobile');
   $usedFields["Mobile"] = true;
}
if (!$usedFields["Other"]) {
   $simplified_xml->writeElement('Office');
   $usedFields["Other"] = true;
}
```
**Describe alternatives you've considered**
I changed it in the code directly, and it works ;-)

The output must look like this:
```
<?xml version="1.0" encoding="UTF-8"?>
<IPPhoneDirectory>
    <DirectoryEntry>
        <Name>John Doe</Name>
        <Telephone>123</Telephone>
        <Office>345</Office>
        <Mobile>123</Mobile>
    </DirectoryEntry>
    <DirectoryEntry>
        <Name>Ray Albert</Name>
        <Telephone>8971</Telephone>
        <Office>1233</Office>
        <Mobile>1233</Mobile>
    </DirectoryEntry>
</IPPhoneDirectory>
```

**Comment by Massi-X** [comment]
Duplicated of #5

**Comment by shamrockDE** [comment]
Hi, it is not a duplicate of #5. The belong together, but are different points.

- This issue #6 is a feature request, to have exactly three entries per name in the XML-Output
- the Issue #5 is a feature request, to have user defined names for the entries

**Comment by Massi-X** [comment]
Hi to have exactly 3 entries in the output you have to enable "phone with limitations" in the gui. If this is the case i will close this and reopen the other one!

**Comment by shamrockDE** [comment]
If i enable "phone with limitations" in the gui, i will have up to 3 entries in the output, so between 1 and 3 entries. Setting the option ensures, that the 3 entries have different names and one entry is definitely called "Telephone". But you can have less than three entries. I tested it several times.
My feature request says, that the output-File has for every name exactly 3 entries. If there are less than 3 numbers in carddav, it will fill up with empty ones. See my coding example.

**Comment by Massi-X** [comment]
Oh sorry I missed that!
So if you choose "phone with limitations" you would like to have all the three fields always present. I think this will require another option because maybe some other phones will break too, but anyway in the example XML posted above I saw you filled the empty entries with fake numbers, is this really needed? Or is it enough to create empty telephone, mobile, office entries?
I don't feel very comfortable creating such dummy entries, they may cause confusion.

**Comment by shamrockDE** [comment]
The result of the above code is like this (i tested it):
```
<?xml version="1.0" encoding="UTF-8"?>
<IPPhoneDirectory>
    <DirectoryEntry>
        <Name>John Doe</Name>
        <Telephone>123</Telephone>
        <Office/>
        <Mobile>123</Mobile>
    </DirectoryEntry>
    <DirectoryEntry>
        <Name>Ray Albert</Name>
        <Telephone>8971</Telephone>
        <Office/>
        <Mobile/>
    </DirectoryEntry>
</IPPhoneDirectory>
```

**Comment by Massi-X** [comment]
Ok perfect will fix it! Thank you

**Comment by shamrockDE** [comment]
To see how it can be i made a fork from your project. I have defined a new phone type ("2"), which stores exactly 3 numbers in the XML-output. And i added new fields in the gui and the config-file, to change the names of the 3 entries in the XML-output (for my phones i have changed "Other" to "Office"). It works in my environement.
I've changed "page.phonemiddleware.php" (for the new names in the gui), "carddavtoXML.php" and backgroundHelper.php (to get the new names from the config file) and "phoneMiddleware.php" (for both changes).
Feel free to look at my solution (https://github.com/shamrockDE/freepbx-phonemiddleware)
